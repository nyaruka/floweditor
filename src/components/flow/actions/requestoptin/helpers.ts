import { getActionUUID } from 'components/flow/actions/helpers';
import { Types } from 'config/interfaces';
import { RequestOptIn } from 'flowTypes';
import { NodeEditorSettings } from 'store/nodeEditor';
import { RequestOptInFormState } from './RequestOptInForm';

export const initializeForm = (settings: NodeEditorSettings): RequestOptInFormState => {
  if (settings.originalAction && settings.originalAction.type === Types.request_optin) {
    const action = settings.originalAction as RequestOptIn;
    return {
      optin: { value: action.optin },
      valid: true
    };
  }

  return {
    optin: { value: null },
    valid: false
  };
};

export const stateToAction = (
  settings: NodeEditorSettings,
  formState: RequestOptInFormState
): RequestOptIn => {
  const result = {
    type: Types.request_optin,
    optin: formState.optin.value[0],
    uuid: getActionUUID(settings, Types.add_input_labels)
  };
  return result;
};
