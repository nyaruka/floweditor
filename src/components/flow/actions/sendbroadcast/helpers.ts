import {
  getActionUUID,
  getComposeActionToState,
  getComposeStateToAction,
  getRecipients,
  getRecipientsStateToAction
} from 'components/flow/actions/helpers';
import { SendBroadcastFormState } from 'components/flow/actions/sendbroadcast/SendBroadcastForm';
import { Types } from 'config/interfaces';
import { BroadcastMsg } from 'flowTypes';
import { NodeEditorSettings } from 'store/nodeEditor';

export const initializeForm = (settings: NodeEditorSettings): SendBroadcastFormState => {
  if (settings.originalAction && settings.originalAction.type === Types.send_broadcast) {
    let action = settings.originalAction as BroadcastMsg;

    // check if our form should use a localized action
    if (settings.localizations && settings.localizations.length > 0) {
      const localized = settings.localizations[0];
      if (localized.isLocalized()) {
        action = settings.localizations[0].getObject() as BroadcastMsg;
      } else {
        return {
          compose: { value: getComposeActionToState() },
          recipients: { value: [] },
          valid: true
        };
      }
    }

    return {
      compose: { value: getComposeActionToState(action) },
      recipients: { value: getRecipients(action) },
      valid: true
    };
  }

  return {
    compose: { value: getComposeActionToState() },
    recipients: { value: [] },
    valid: false
  };
};

export const stateToAction = (
  settings: NodeEditorSettings,
  formState: SendBroadcastFormState
): BroadcastMsg => {
  const [compose, text, attachments] = getComposeStateToAction(formState);
  const [legacy_vars, contacts, groups] = getRecipientsStateToAction(formState);

  return {
    legacy_vars: legacy_vars,
    contacts: contacts,
    groups: groups,
    compose: compose,
    text: text,
    attachments: attachments,
    type: Types.send_broadcast,
    uuid: getActionUUID(settings, Types.send_broadcast)
  };
};
