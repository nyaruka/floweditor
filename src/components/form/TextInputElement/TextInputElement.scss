@import '../../../variables.scss';

.count {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: end;
        -ms-flex-pack: end;
            justify-content: flex-end;
    margin: 5px 0;
}

.tooltip {
    position: relative;
    display: inline-block;
    font-size: 15px;
    margin-left: 5px;
}

.tooltip .tooltiptext {
    visibility: hidden;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
        -ms-flex-direction: column;
            flex-direction: column;
    -webkit-box-pack: justify;
        -ms-flex-pack: justify;
            justify-content: space-between;
    height: 220px;
    width: 175px;
    background-color: white;
    color: $grey;
    text-align: left;
    padding: 12px;
    border-radius: 4px;
    font-size: 14px;
    line-height: 20px;
    -webkit-box-shadow: $shadow_down;
    box-shadow: $shadow_down;

    /* Position tooltip text */
    position: absolute;
    z-index: 1;
    bottom: 138%;
    left: -490%;
    margin-left: -60px;

    /* Fade in tooltip */
    opacity: 0;
    -webkit-transition: opacity 0.25s;
    -o-transition: opacity 0.25s;
    transition: opacity 0.25s;
}

/* Show tooltip text when mouse hovers over the tooltip container */
.tooltip:hover .tooltiptext {
    visibility: visible;
    opacity: 1;
}

.tooltiprow {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
        -ms-flex-pack: justify;
            justify-content: space-between;
}

.option_list {
    overflow-y: auto;
    list-style-type: none;
    padding: 0;
    max-height: 300px;
    margin-bottom: 0px;
}

.option_description {
    font-size: 12px;
    padding: 3px 0;
    font-weight: 300;
}

.option {
    border-top: 1px solid transparent;
    border-bottom: 1px solid transparent;
    padding: 5px;
    font-size: 14px;
    &.selected {
        border-color: $blue_antd;
        background: $blue_antd;
        color: $text_color;
        font-weight: 500;
        &.first_option {
            border-top-color: transparent !important;
        }
    }
}

.completion_container {
    position: absolute;
    background: #fff;
    margin-top: 20px;
    width: 250px;
    border-radius: 4px;
    -webkit-box-shadow: $shadow_down;
            box-shadow: $shadow_down;
    z-index: $z_completion;
}

.wrapper {
    position: relative;
}

.help {
    font-size: 10px;
    color: #999;
    padding: 4px;
    border-radius: 0 0 4px 4px;
    background-color: #f5f5f5;
}

.hidden {
    display: none;
}

.textinput {
    border-radius: 4px;
    width: 100%;
    color: $grey;
    font-size: 14px;
    font-weight: 300;
    border: 1px solid $light_grey;
    overflow: none;
    outline: none;
    -webkit-box-shadow: none;
    box-shadow: none;
    padding: 10px 10px;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    height: 37px;
    &::-webkit-input-placeholder {
        color: $placeholder_color;
    }
    &::-moz-placeholder {
        color: $placeholder_color;
    }
    &:-ms-input-placeholder {
        color: $placeholder_color;
    }
    &:-moz-placeholder {
        color: $placeholder_color;
    }
}

textarea.textinput {
    height: 120px;
    padding: 10px;
    overflow: auto;
}
