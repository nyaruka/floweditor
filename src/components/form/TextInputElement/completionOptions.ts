import { CompletionOption } from '../../../services/ComponentMap';

export const OPTIONS: CompletionOption[] = [
    { name: 'contact', description: 'The name of the contact.' },
    { name: 'contact.name', description: 'The name of the contact.' },
    { name: 'contact.language', description: 'The language code for the contact.' },
    { name: 'contact.fields', description: 'Custom fields on the contact.' },
    { name: 'contact.groups', description: 'The groups for the contact.' },
    { name: 'contact.urns', description: 'URNs on the contact.' },
    { name: 'contact.urns.tel', description: 'The preferred telephone number for the contact.' },
    { name: 'contact.urns.telegram', description: 'The preferred telegram id for the contact.' },
    { name: 'input', description: 'The last input from the contact if any.' },
    { name: 'run', description: 'Run details' },
    { name: 'run.contact', description: 'The contact in this run' },
    { name: 'run.results', description: 'Results for the run' },
    { name: 'child', description: 'Run details after running a child flow' },
    { name: 'child.results', description: 'The results for the child flow' },
    { name: 'parent', description: 'Run details if being called from a parent flow' },
    { name: 'parent.results', description: 'The results for the parent flow' },
    { name: 'webhook', description: 'The body of the webhook response' },
    { name: 'webhook.status', description: 'The status of the webhook call' },
    { name: 'webhook.status_code', description: 'The status code returned from the webhook' },
    { name: 'webhook.url', description: 'The URL which was called' },
    { name: 'webhook.body', description: 'The body of the webhook response' },
    {
        name: 'webhook.json',
        description: 'The JSON parsed body of the response, can access subelements'
    },
    { name: 'webhook.request', description: 'The raw request of the webhook including headers' },
    { name: 'webhook.response', description: 'The raw response of the webhook including headers' }
];
