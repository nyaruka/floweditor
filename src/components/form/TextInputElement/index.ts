import TextInputElement, { Count, CharacterSet, HTMLTextElement } from './TextInputElement';

export { Count, CharacterSet, HTMLTextElement };

export default TextInputElement;
