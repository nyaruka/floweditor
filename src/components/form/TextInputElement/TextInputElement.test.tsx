import * as React from 'react';
import { shallow } from 'enzyme';
import { getSpecWrapper } from '../../../helpers/utils';
import ComponentMap from '../../../services/ComponentMap';
import TextInputElement, {
    Count,
    CharacterSet,
    MAX_GSM_SINGLE,
    MAX_GSM_MULTI,
    MAX_UNICODE_SINGLE,
    MAX_UNICODE_MULTI
} from './TextInputElement';

const {
    results: [{ definition }]
} = require('../../../../test_flows/a4f64f1b-85bc-477e-b706-de313a022979.json');

const CompMap = new ComponentMap(definition);
const props = {
    name: 'Message',
    count: Count.SMS,
    showLabel: false,
    placeholder: '',
    autocomplete: true,
    required: true,
    textarea: true,
    ComponentMap: CompMap
};

describe('Component: TextInputElement', () => {
    it('should recognize GSM 7-bit characters and maintain count accordingly', () => {
        let TextInputGSM = shallow(
            <TextInputElement
                {...{
                    ...props,
                    value:
                        "What's your favorite color, (r)ed, (o)range, (y)ellow, (g)reen, (b)lue, (i)ndigo or (v)iolet?"
                }}
            />
        );

        expect(TextInputGSM.state('maxLength')).toBe(MAX_GSM_SINGLE);
        expect(TextInputGSM.state('parts').length).toBe(1);
        expect(TextInputGSM.state('characterSet')).toBe(CharacterSet.GSM);
        expect(TextInputGSM.state('remainingInPart')).toBe(67);
        expect(TextInputGSM.state('characterCount')).toBe(93);
        expect(
            getSpecWrapper(TextInputGSM, 'counter')
                .text()
                .indexOf('67/1 ?')
        ).toBe(0);

        TextInputGSM = shallow(
            <TextInputElement
                {...{
                    ...props,
                    value:
                        "What's your favorite color, (r)ed, (o)range, (y)ellow, (g)reen, (b)lue, (i)ndigo or (v)iolet? { Fun fact: Approximately seven million different colors can be seen by the human eye. }"
                }}
            />
        );

        expect(TextInputGSM.state('maxLength')).toBe(MAX_GSM_MULTI);
        expect(TextInputGSM.state('parts').length).toBe(2);
        expect(TextInputGSM.state('characterSet')).toBe(CharacterSet.GSM);
        expect(TextInputGSM.state('characterCount')).toBe(182);
        expect(TextInputGSM.state('remainingInPart')).toBe(122);
        expect(
            getSpecWrapper(TextInputGSM, 'counter')
                .text()
                .indexOf('122/2 ?')
        ).toBe(0);
    });

    it('should recognize unicode characters and maintain count accordingly', () => {
        let TextInputUni = shallow(
            <TextInputElement
                {...{
                    ...props,
                    value: "ðŸ‘‹ Hi, what's your name?"
                }}
            />
        );

        expect(TextInputUni.state('maxLength')).toBe(MAX_UNICODE_SINGLE);
        expect(TextInputUni.state('parts').length).toBe(1);
        expect(TextInputUni.state('characterSet')).toBe(CharacterSet.UNICODE);
        expect(TextInputUni.state('characterCount')).toBe(23);
        expect(TextInputUni.state('remainingInPart')).toBe(46);
        expect(
            getSpecWrapper(TextInputUni, 'counter')
                .text()
                .indexOf('46/1 ?')
        ).toBe(0);

        TextInputUni = shallow(
            <TextInputElement
                {...{
                    ...props,
                    value:
                        "ðŸ‘‹ Hi, what's your name? We're doing a survey in your local community and would appreciate your help!"
                }}
            />
        );

        expect(TextInputUni.state('maxLength')).toBe(MAX_UNICODE_MULTI);
        expect(TextInputUni.state('parts').length).toBe(2);
        expect(TextInputUni.state('characterSet')).toBe(CharacterSet.UNICODE);
        expect(TextInputUni.state('remainingInPart')).toBe(33);
        expect(TextInputUni.state('characterCount')).toBe(100);
        expect(
            getSpecWrapper(TextInputUni, 'counter')
                .text()
                .indexOf('33/2 ?')
        ).toBe(0);
    });
});
