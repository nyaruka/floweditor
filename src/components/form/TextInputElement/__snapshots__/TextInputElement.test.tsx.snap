// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputElement > helpers > filterOptions > should filter options for "c" 1`] = `
Array [
  Object {
    "description": "The name of the contact.",
    "name": "contact",
  },
  Object {
    "description": "Run details after running a child flow",
    "name": "child",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "ch" 1`] = `
Array [
  Object {
    "description": "Run details after running a child flow",
    "name": "child",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "child" 1`] = `
Array [
  Object {
    "description": "Run details after running a child flow",
    "name": "child",
  },
  Object {
    "description": "The results for the child flow",
    "name": "child.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "child." 1`] = `
Array [
  Object {
    "description": "The results for the child flow",
    "name": "child.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "child.r" 1`] = `
Array [
  Object {
    "description": "The results for the child flow",
    "name": "child.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "child.re" 1`] = `
Array [
  Object {
    "description": "The results for the child flow",
    "name": "child.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "child.results" 1`] = `
Array [
  Object {
    "description": "The results for the child flow",
    "name": "child.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "co" 1`] = `
Array [
  Object {
    "description": "The name of the contact.",
    "name": "contact",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact" 1`] = `
Array [
  Object {
    "description": "The name of the contact.",
    "name": "contact",
  },
  Object {
    "description": "The name of the contact.",
    "name": "contact.name",
  },
  Object {
    "description": "The language code for the contact.",
    "name": "contact.language",
  },
  Object {
    "description": "Custom fields on the contact.",
    "name": "contact.fields",
  },
  Object {
    "description": "The groups for the contact.",
    "name": "contact.groups",
  },
  Object {
    "description": "URNs on the contact.",
    "name": "contact.urns",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact." 1`] = `
Array [
  Object {
    "description": "The name of the contact.",
    "name": "contact.name",
  },
  Object {
    "description": "The language code for the contact.",
    "name": "contact.language",
  },
  Object {
    "description": "Custom fields on the contact.",
    "name": "contact.fields",
  },
  Object {
    "description": "The groups for the contact.",
    "name": "contact.groups",
  },
  Object {
    "description": "URNs on the contact.",
    "name": "contact.urns",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.f" 1`] = `
Array [
  Object {
    "description": "Custom fields on the contact.",
    "name": "contact.fields",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.fi" 1`] = `
Array [
  Object {
    "description": "Custom fields on the contact.",
    "name": "contact.fields",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.fields" 1`] = `
Array [
  Object {
    "description": "Custom fields on the contact.",
    "name": "contact.fields",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.g" 1`] = `
Array [
  Object {
    "description": "The groups for the contact.",
    "name": "contact.groups",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.gr" 1`] = `
Array [
  Object {
    "description": "The groups for the contact.",
    "name": "contact.groups",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.groups" 1`] = `
Array [
  Object {
    "description": "The groups for the contact.",
    "name": "contact.groups",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.l" 1`] = `
Array [
  Object {
    "description": "The language code for the contact.",
    "name": "contact.language",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.la" 1`] = `
Array [
  Object {
    "description": "The language code for the contact.",
    "name": "contact.language",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.language" 1`] = `
Array [
  Object {
    "description": "The language code for the contact.",
    "name": "contact.language",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.n" 1`] = `
Array [
  Object {
    "description": "The name of the contact.",
    "name": "contact.name",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.na" 1`] = `
Array [
  Object {
    "description": "The name of the contact.",
    "name": "contact.name",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.name" 1`] = `
Array [
  Object {
    "description": "The name of the contact.",
    "name": "contact.name",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.u" 1`] = `
Array [
  Object {
    "description": "URNs on the contact.",
    "name": "contact.urns",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.ur" 1`] = `
Array [
  Object {
    "description": "URNs on the contact.",
    "name": "contact.urns",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.urns" 1`] = `
Array [
  Object {
    "description": "URNs on the contact.",
    "name": "contact.urns",
  },
  Object {
    "description": "The preferred telephone number for the contact.",
    "name": "contact.urns.tel",
  },
  Object {
    "description": "The preferred telegram id for the contact.",
    "name": "contact.urns.telegram",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.urns.t" 1`] = `
Array [
  Object {
    "description": "The preferred telephone number for the contact.",
    "name": "contact.urns.tel",
  },
  Object {
    "description": "The preferred telegram id for the contact.",
    "name": "contact.urns.telegram",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.urns.te" 1`] = `
Array [
  Object {
    "description": "The preferred telephone number for the contact.",
    "name": "contact.urns.tel",
  },
  Object {
    "description": "The preferred telegram id for the contact.",
    "name": "contact.urns.telegram",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.urns.tel" 1`] = `
Array [
  Object {
    "description": "The preferred telephone number for the contact.",
    "name": "contact.urns.tel",
  },
  Object {
    "description": "The preferred telegram id for the contact.",
    "name": "contact.urns.telegram",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "contact.urns.telegram" 1`] = `
Array [
  Object {
    "description": "The preferred telegram id for the contact.",
    "name": "contact.urns.telegram",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "i" 1`] = `
Array [
  Object {
    "description": "The last input from the contact if any.",
    "name": "input",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "in" 1`] = `
Array [
  Object {
    "description": "The last input from the contact if any.",
    "name": "input",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "input" 1`] = `
Array [
  Object {
    "description": "The last input from the contact if any.",
    "name": "input",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "input." 1`] = `Array []`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "p" 1`] = `
Array [
  Object {
    "description": "Run details if being called from a parent flow",
    "name": "parent",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "pa" 1`] = `
Array [
  Object {
    "description": "Run details if being called from a parent flow",
    "name": "parent",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "parent" 1`] = `
Array [
  Object {
    "description": "Run details if being called from a parent flow",
    "name": "parent",
  },
  Object {
    "description": "The results for the parent flow",
    "name": "parent.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "parent." 1`] = `
Array [
  Object {
    "description": "The results for the parent flow",
    "name": "parent.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "parent.r" 1`] = `
Array [
  Object {
    "description": "The results for the parent flow",
    "name": "parent.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "parent.re" 1`] = `
Array [
  Object {
    "description": "The results for the parent flow",
    "name": "parent.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "parent.results" 1`] = `
Array [
  Object {
    "description": "The results for the parent flow",
    "name": "parent.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "r" 1`] = `
Array [
  Object {
    "description": "Run details",
    "name": "run",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "ru" 1`] = `
Array [
  Object {
    "description": "Run details",
    "name": "run",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "run" 1`] = `
Array [
  Object {
    "description": "Run details",
    "name": "run",
  },
  Object {
    "description": "The contact in this run",
    "name": "run.contact",
  },
  Object {
    "description": "Results for the run",
    "name": "run.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "run." 1`] = `
Array [
  Object {
    "description": "The contact in this run",
    "name": "run.contact",
  },
  Object {
    "description": "Results for the run",
    "name": "run.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "run.c" 1`] = `
Array [
  Object {
    "description": "The contact in this run",
    "name": "run.contact",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "run.co" 1`] = `
Array [
  Object {
    "description": "The contact in this run",
    "name": "run.contact",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "run.contact" 1`] = `
Array [
  Object {
    "description": "The contact in this run",
    "name": "run.contact",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "run.r" 1`] = `
Array [
  Object {
    "description": "Results for the run",
    "name": "run.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "run.re" 1`] = `
Array [
  Object {
    "description": "Results for the run",
    "name": "run.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "run.results" 1`] = `
Array [
  Object {
    "description": "Results for the run",
    "name": "run.results",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "w" 1`] = `
Array [
  Object {
    "description": "The body of the webhook response",
    "name": "webhook",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "we" 1`] = `
Array [
  Object {
    "description": "The body of the webhook response",
    "name": "webhook",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook" 1`] = `
Array [
  Object {
    "description": "The body of the webhook response",
    "name": "webhook",
  },
  Object {
    "description": "The status of the webhook call",
    "name": "webhook.status",
  },
  Object {
    "description": "The status code returned from the webhook",
    "name": "webhook.status_code",
  },
  Object {
    "description": "The URL which was called",
    "name": "webhook.url",
  },
  Object {
    "description": "The body of the webhook response",
    "name": "webhook.body",
  },
  Object {
    "description": "The JSON parsed body of the response, can access subelements",
    "name": "webhook.json",
  },
  Object {
    "description": "The raw request of the webhook including headers",
    "name": "webhook.request",
  },
  Object {
    "description": "The raw response of the webhook including headers",
    "name": "webhook.response",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook." 1`] = `
Array [
  Object {
    "description": "The status of the webhook call",
    "name": "webhook.status",
  },
  Object {
    "description": "The status code returned from the webhook",
    "name": "webhook.status_code",
  },
  Object {
    "description": "The URL which was called",
    "name": "webhook.url",
  },
  Object {
    "description": "The body of the webhook response",
    "name": "webhook.body",
  },
  Object {
    "description": "The JSON parsed body of the response, can access subelements",
    "name": "webhook.json",
  },
  Object {
    "description": "The raw request of the webhook including headers",
    "name": "webhook.request",
  },
  Object {
    "description": "The raw response of the webhook including headers",
    "name": "webhook.response",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.b" 1`] = `
Array [
  Object {
    "description": "The body of the webhook response",
    "name": "webhook.body",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.bo" 1`] = `
Array [
  Object {
    "description": "The body of the webhook response",
    "name": "webhook.body",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.body" 1`] = `
Array [
  Object {
    "description": "The body of the webhook response",
    "name": "webhook.body",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.j" 1`] = `
Array [
  Object {
    "description": "The JSON parsed body of the response, can access subelements",
    "name": "webhook.json",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.js" 1`] = `
Array [
  Object {
    "description": "The JSON parsed body of the response, can access subelements",
    "name": "webhook.json",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.json" 1`] = `
Array [
  Object {
    "description": "The JSON parsed body of the response, can access subelements",
    "name": "webhook.json",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.r" 1`] = `
Array [
  Object {
    "description": "The raw request of the webhook including headers",
    "name": "webhook.request",
  },
  Object {
    "description": "The raw response of the webhook including headers",
    "name": "webhook.response",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.re" 1`] = `
Array [
  Object {
    "description": "The raw request of the webhook including headers",
    "name": "webhook.request",
  },
  Object {
    "description": "The raw response of the webhook including headers",
    "name": "webhook.response",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.request" 1`] = `
Array [
  Object {
    "description": "The raw request of the webhook including headers",
    "name": "webhook.request",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.response" 1`] = `
Array [
  Object {
    "description": "The raw response of the webhook including headers",
    "name": "webhook.response",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.s" 1`] = `
Array [
  Object {
    "description": "The status of the webhook call",
    "name": "webhook.status",
  },
  Object {
    "description": "The status code returned from the webhook",
    "name": "webhook.status_code",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.st" 1`] = `
Array [
  Object {
    "description": "The status of the webhook call",
    "name": "webhook.status",
  },
  Object {
    "description": "The status code returned from the webhook",
    "name": "webhook.status_code",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.status" 1`] = `
Array [
  Object {
    "description": "The status of the webhook call",
    "name": "webhook.status",
  },
  Object {
    "description": "The status code returned from the webhook",
    "name": "webhook.status_code",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.status_code" 1`] = `
Array [
  Object {
    "description": "The status code returned from the webhook",
    "name": "webhook.status_code",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.u" 1`] = `
Array [
  Object {
    "description": "The URL which was called",
    "name": "webhook.url",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.ur" 1`] = `
Array [
  Object {
    "description": "The URL which was called",
    "name": "webhook.url",
  },
]
`;

exports[`TextInputElement > helpers > filterOptions > should filter options for "webhook.url" 1`] = `
Array [
  Object {
    "description": "The URL which was called",
    "name": "webhook.url",
  },
]
`;

exports[`TextInputElement > helpers > getCharCount > should generate character count stats for msg of type "GSM: 7-bit" 1`] = `
Object {
  "characterCount": 24,
  "characterSet": "GSM",
  "maxLength": 160,
  "parts": Array [
    Object {
      "bytes": 24,
      "content": "GSM, regular 7-bit chars",
      "length": 24,
    },
  ],
  "remainingInPart": 136,
  "value": "GSM, regular 7-bit chars",
}
`;

exports[`TextInputElement > helpers > getCharCount > should generate character count stats for msg of type "GSM: escape" 1`] = `
Object {
  "characterCount": 27,
  "characterSet": "GSM",
  "maxLength": 160,
  "parts": Array [
    Object {
      "bytes": 35,
      "content": "GSM, escape chars: |^€{}[]~",
      "length": 27,
    },
  ],
  "remainingInPart": 125,
  "value": "GSM, escape chars: |^€{}[]~",
}
`;

exports[`TextInputElement > helpers > getCharCount > should generate character count stats for msg of type "Unicode + GSM escape" 1`] = `
Object {
  "characterCount": 38,
  "characterSet": "Unicode",
  "maxLength": 70,
  "parts": Array [
    Object {
      "bytes": 78,
      "content": "Unicode 💩 w/ GSM escape chars |^€{}[]~",
      "length": 38,
    },
  ],
  "remainingInPart": 31,
  "value": "Unicode 💩 w/ GSM escape chars |^€{}[]~",
}
`;

exports[`TextInputElement > helpers > getCharCount > should generate character count stats for msg of type "Unicode" 1`] = `
Object {
  "characterCount": 9,
  "characterSet": "Unicode",
  "maxLength": 70,
  "parts": Array [
    Object {
      "bytes": 20,
      "content": "Unicode 💩",
      "length": 9,
    },
  ],
  "remainingInPart": 60,
  "value": "Unicode 💩",
}
`;

exports[`TextInputElement > helpers > getCharCountStats > should return an object containing character count stats if passed Count.SMS enum 1`] = `
Object {
  "characterCount": 24,
  "characterSet": "GSM",
  "maxLength": 160,
  "parts": Array [
    Object {
      "bytes": 24,
      "content": "GSM, regular 7-bit chars",
      "length": 24,
    },
  ],
  "remainingInPart": 136,
  "value": "GSM, regular 7-bit chars",
}
`;
